src/server/services/article.service.ts

```diff
@@ -410,11 +407,12 @@ export const getArticles = async ({
       select: { ...profileImageSelect, ingestion: true },
     });
 
+    const coverImages = await dbRead.image.findMany({
+      where: { id: { in: articles.map((x) => x.coverId).filter(isDefined) } },
+      select: imageSelect,
+    });
```

Comment:
We might want to check whether there are coverIds or not to avoid making a request to the db with an empty array, which would return all images if I'm not mistaken
```suggestion
    const coverIds = articles.map((x) => x.coverId).filter(isDefined);
    const coverImages = coverIds.length ? await dbRead.image.findMany({
      where: { id: { in: coverIds } },
      select: imageSelect,
    }) : [];
```